crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -169,6 +169,15 @@ impl<'a> Checker<'a> {
         )
     }
 
+    /// Returns the [`Isolation`] level for fixes in the current context.
+    pub(crate) fn isolation(&self, parent: Option<&Stmt>) -> Isolation {
```

Comment:
Could you explain this in more detail? Should I use this for all fixes or just some fixes that apply to a certain parent? It may also be worth renaming the method to make it more explicit, what it's use case is (when to use it)